test/tools/ossfuzz/protoToAbiV2.cpp

```diff
@@ -730,8 +841,12 @@ void ProtoConverter::writeHelperFunctions()
 		internal pure returns (bytes memory, bytes memory)
 	{
 		bytes memory validEncoding = new bytes(4 + argumentEncoding.length);
-		// Ensure that invalidEncoding crops at least one and at most all bytes from correct encoding.
-		uint invalidLength = invalidLengthFuzz % argumentEncoding.length;
+		// Ensure that invalidEncoding crops at least 32 bytes (padding length
+		// is at most 31 bytes) since shorter bytes/string values can lead to
```

Comment:
We could actually do this conditionally, i.e. only if the last argument of the function is in fact ``bytes`` or ``string``... Would that information be easily available here? If not it's not too important.

Reply:
Okay, I can do that.

Reply:
I'm wondering if we would need to crop at least 32 bytes if the last argument to the function is an array. May be I should simply read the abi encoder specification :-)

Reply:
For reference, here is what the encoder spec says

> uint<M>: enc(X) is the big-endian encoding of X, padded on the higher-order (left) side with zero-bytes such that the length is 32 bytes.

> int<M>: enc(X) is the big-endian two’s complement encoding of X, padded on the higher-order (left) side with 0xff for negative X and with zero bytes for positive X such that the length is 32 bytes.

So doesn't that mean we may have to crop at least 31 bytes for (u)intX as well.

Also, since encoding of an array are composed of encoding of the base type, doesn't it mean we would need to extend the cropping strategy to static/dynamic
  - arrays of bytes/string?
  - arrays of (u)intX?

Reply:
Ah, just realized my mistake. (u)intX are left padded. Only bytes/string are right padded. So if I understood correctly, we should conditionally chop off at least 32 bytes when the last argument to the function is one of the following types:
  - bytes/string
  - base type of a static/dynamic array is bytes/string

Reply:
@ekpyron Is my assessment here correct? Basically, the addition to what you requested above (https://github.com/ethereum/solidity/pull/7188#discussion_r312000815) is that we crop at least 32 bytes for arrays whose base type is bytes/string.

Reply:
I think so - if the last argument is ``bytes`` or ``string`` itself or an array with one of these as last base type, then encodings without padding at the end are fine, so that's the cases in which we have to crop at least 32 bytes. I think that's fine!